source/tutorials/learning-journey/sharing-dependencies.md

```diff
@@ -0,0 +1,204 @@
+
+### How long will it take?
+
+This tutorial will take approximately 1 hour.
+
+### What will you need?
+
+This tutorial assumes you've seen a derivation (`mkDerivation`, `buildPythonApplication`, etc) before, and that you've seen `nix-shell` used to create shell environments.
+While this tutorial uses Python as the language for the example project, no actual Python knowledge is requried.
+
+## Setting the stage
+
+Suppose you have a working build for your project in a `default.nix` file so that when you run `nix-build` it builds your project.
+It includes all of the dependencies needed to build it, but nothing more.
+Now suppose you wanted to bring in some tools during development, such as a linter, a code formatter, [git commit hooks][git_hooks], etc.
+
+[git_hooks]: https://github.com/cachix/pre-commit-hooks.nix
+
+One solution could be to add those packages to your build.
+This would certainly work in a pinch, but now your build depends on packages that aren't necessary for it to actually build.
+A better solution is to add those development packages to a shell environment so that the build dependencies stay as lean as possible.
+
+However, now you need to define a `shell.nix` that not only provides your development packages, but can also build your project.
+In other words, you need a `shell.nix` that brings in all of the packages that your build depends on.
+You could certainly copy the build dependencies from `default.nix` and copy them into `shell.nix`, but this is less than ideal:
+your build dependencies are defined in multiple places, and aside from repeating yourself there's now the possiblity that the dependencies in `default.nix` and `shell.nix` may fall out of sync.
+
+There is a better way!
+
+## Getting started
+
+Create a directory called `shared_project` and enter it:
+
+```console
+$ mkdir shared_project
+$ cd shared_project
+```
+
+You'll be creating a Python web application as an example project, but don't worry, you'll be given all of the code you need and won't need to know Python to proceed.
+
+Create a new directory called `src` and two empty files inside of `src` called `__init__.py` and `app.py`:
+
+```
+$ mkdir src
+$ touch src/__init__.py
+$ touch src/app.py
+```
+
+Copy the following contents into `app.py`:
+
+```python
+from flask import Flask
+
+app = Flask(__name__)
+
+@app.route("/")
+def hello_world():
+    return "<p>Hello, World!</p>"
+```
+
+This creates a web application that returns `<p>Hello, World!</p>` on the `/` route.
+
+Next create a `pyproject.toml` file with the following contents:
+
+```toml
+[build-system]
+requires = ["setuptools", "setuptools-scm"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "shared_project"
+version = "0.0.1"
+
+[project.scripts]
+app = "app:main"
+```
+
+This file tells Python how to build the project and what will execute when you run the executable called `app`.
+
+For the Nix part of the project you'll create two files: `build.nix` and `default.nix`.
```

Comment:
```suggestion
For the Nix part of the project you'll create two files: `package.nix` and `default.nix`.
```

We may want to call it `package.nix` or maybe `shared_project.nix`, because that file essentially signifies that particular package.